test/io/test_io.py

```diff
@@ -1164,9 +1178,9 @@ def test_log_postgrest_version(defaultenv):
     with run(env=defaultenv, no_startup_stdout=False) as postgrest:
         version = postgrest.session.head("/").headers["Server"].split("/")[1]
 
-        output = sorted(postgrest.read_stdout(nlines=5))
+        output = postgrest.read_stdout(nlines=1)
 
-        assert "Starting PostgREST %s..." % version in output[4]
+        assert "Starting PostgREST %s..." % version in output[0]
```

Comment:
Since this test only cares about the first line in the output, I don't think it's necessary to check for more than one line and then order them. Sometimes, the "Listening for notifications..." message does not appear until later, messing up the `sort()` done and failing the test.

> Needs work on tests, locally they sometimes fail on `test_log_postgrest_version` or `test_pool_acquisition_timeout`.

This change fixes the `test_log_postgrest_version`. The other one seems related to #3424.